src/vivarium/framework/lookup/table.py

```diff
@@ -209,6 +291,11 @@ def __init__(
         # Note datetime catches pandas timestamps
         if isinstance(data, (Number, datetime, timedelta, list, tuple)):
             self._table = ScalarTable(data, value_columns)
+        elif parameter_columns is None:
```

Comment:
I'm kinda confused by this. Can you explain the logic in this if block? How is it that if `data` is not Number, datetime, timedelta, list, or tuple and `parameter_columns` is None means that this is a categorical table?

Reply:
If the data is a Number, datetime, timedelta (ScalarValue types in lookup.py) or a list or tuple of these types, it can be built using a ScalarTable. Otherwise, it's a DataFrame (the type checking for data in the init of LookupTable allows for Union[ScalarValue, pd.DataFrame, List[ScalarValue], Tuple[ScalarValue]]). 

The DataFrame will have parameter columns or key columns (or both). Parameter columns contain continuous values and key columns contain categorical values. So if we have no parameter columns we only have key columns aka categorical values.